admin_app/src/app/playground/components/PlaygroundComponents.tsx

```diff
@@ -310,17 +360,48 @@ const MessageBox = (message: Message) => {
             ? message.content
             : renderResults(message.content)}
         </Typography>
-        {message.hasOwnProperty("json") && (
-          <Link
-            onClick={toggleJsonModal}
-            variant="caption"
-            align="right"
-            underline="hover"
-            sx={{ cursor: "pointer" }}
+        {message.type == "response" ? (
+          <Box
+            style={{
+              marginTop: "5px",
+              display: "flex",
+              justifyContent: "flex-end",
+              alignItems: "center",
+            }}
           >
-            {"<json>"}
-          </Link>
-        )}
+            {message.json.hasOwnProperty("feedback_secret_key") ? (
+              <Box sx={{ marginRight: "8px" }}>
+                <IconButton
+                  aria-label="thumbs up"
+                  onClick={() => handleFeedback("positive")}
+                  style={feedbackButtonStyle}
+                >
+                  {feedbackMapping.positive.state == true
+                    ? feedbackMapping.positive.onIcon
+                    : feedbackMapping.positive.offIcon}
+                </IconButton>
+                <IconButton
+                  aria-label="thumbs down"
+                  onClick={() => handleFeedback("negative")}
+                  style={feedbackButtonStyle}
+                >
+                  {feedbackMapping.negative.state == true
+                    ? feedbackMapping.negative.onIcon
+                    : feedbackMapping.negative.offIcon}
+                </IconButton>
```

Comment:
Uses `FeedbacMapping`

Just realized this adds another layer of nested ternary operation 😓

Reply:
I guess you could iterate over feedbackMapping to create each of these? But don't spend too much time on this.